### PR TITLE
Fix cargo audit failure

### PR DESCRIPTION
**Pull request by julesdesmit**

### CODE REVIEW

**Comment by ibaryshnikov** [comment]
@julesdesmit thanks for taking care of this warning! We may also need to fix this for `phase2` branch.

File: phase1-coordinator/src/coordinator.rs [hunk 1]
```diff
@@ -30,11 +30,11 @@ use crate::{
 };
 use setup_utils::calculate_hash;
 
-use chrono::{DateTime, Utc};
 use std::{
     fmt,
     sync::{Arc, RwLock},
 };
+use time::PrimitiveDateTime;
```
Comment (niklaslong) [inline phase1-coordinator/src/coordinator.rs:37]: Why not `OffsetDateTime`?

File: phase1-coordinator/src/coordinator.rs [hunk 2]
```diff
@@ -295,39 +295,39 @@ impl SystemTimeSource {
 }
 
 impl TimeSource for SystemTimeSource {
-    fn utc_now(&self) -> DateTime<Utc> {
-        Utc::now()
+    fn utc_now(&self) -> PrimitiveDateTime {
+        PrimitiveDateTime::now()
```
Comment (niklaslong) [inline phase1-coordinator/src/coordinator.rs:299]: This method is deprecated, how about `OffsetDateTime::now_utc`? https://time-rs.github.io/api/time/struct.OffsetDateTime.html#method.now_utc

File: phase1-coordinator/src/coordinator_state.rs [hunk 3]
```diff
@@ -2625,11 +2625,11 @@ impl CoordinatorState {
...
                 if elapsed > contributor_seen_timeout && !self.is_coordinator_contributor(&participant) {
                     tracing::warn!(
-                        "Dropping participant {} because it has exceeded the maximum ({}) allowed time \
-                        since it was last seen by the coordinator (last seen {} ago).",
+                        "Dropping participant {} because it has exceeded the maximum ({:?}) allowed time \
+                        since it was last seen by the coordinator (last seen {:?} ago).",
                         participant,
-                        chrono_humanize::HumanTime::from(contributor_seen_timeout),
-                        chrono_humanize::HumanTime::from(elapsed)
+                        contributor_seen_timeout,
```
Comment (niklaslong) [inline phase1-coordinator/src/coordinator_state.rs:2631]: Couldn't really find a suitable replacement for `chrono-humanize` when I looked into this 👍

File: phase1-coordinator/src/environment.rs [hunk 4]
```diff
@@ -225,21 +224,17 @@ pub struct Environment {
     /// Returns the maximum duration a contributor can go without
     /// being seen by the coordinator before it will be dropped from
     /// the ceremony by the coordinator.
-    #[serde_as(as = "DurationSecondsWithFrac<String>")]
-    contributor_seen_timeout: chrono::Duration,
+    contributor_seen_timeout: time::Duration,
     /// The maximum duration a verifier can go without being seen by
     /// the coordinator before it will be dropped from the ceremony by
     /// the coordinator.
-    #[serde_as(as = "DurationSecondsWithFrac<String>")]
```
Comment (niklaslong) [inline phase1-coordinator/src/environment.rs:233]: This broke some of the tests when I tried it, not sure if we should be serialising using the `serde-human-readable` feature flag on the `time` crate?

File: setup-utils/Cargo.toml [hunk 5]
```diff
@@ -28,8 +28,8 @@ hex = { version = "0.4.3" }
 num_cpus = { version = "1.12.0" }
 rand = { version = "0.8" }
 rand_chacha = { version = "0.3" }
+sha2 = "0.9"
```
Comment (niklaslong) [inline setup-utils/Cargo.toml:31]: 👍

**Review by niklaslong — COMMENTED** [review]
Looks like there are still a few things to tweak but overall this looks good. I've left a few comments for consideration.

**Comment by apruden2008** [comment]
@julesdesmit can we merge this in?

**Comment by julesdesmit** [comment]
@apruden2008 I believe Niklas' PR actually already fixed the issues with `chrono` at the very least. We sadly can't get past the failing audit check due to most of our dependencies also using vulnerable crates. Besides, this PR is pretty old now and should probably be closed.

**Pull request closed without merging**